targets/atsamd21e18a.json

```diff
@@ -9,5 +9,7 @@
 	"linkerscript": "targets/atsamd21.ld",
 	"extra-files": [
 		"src/device/sam/atsamd21e18a.s"
-	]
+	],
+	"openocd-transport": "swd",
+	"openocd-target": "at91samdXX"
```

Comment:
Have you confirmed that `at91samdXX` indeed works on the ATsamd21, for example by starting a debugger with `tinygo gdb`?

Reply:
### feather-m4

```
$ tinygo gdb --target feather-m4 --programmer cmsis-dap --size short --opt 1 ./src/examples/blinky1
   code    data     bss |   flash     ram
  11360      16    6344 |   11376    6360
GNU gdb (Ubuntu 9.1-0ubuntu1) 9.1
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /tmp/tinygo822515807/main...
Remote debugging using :3333
0x000056ee in machine.sendDescriptor (
    setup=<error reading variable: access outside bounds of object referenced via synthetic pointer>)
    at /home/sago35/tinygo/tinygo/src/machine/usb.go:628
628                     return
Loading section .text, size 0x2c60 lma 0x4000
Loading section .data, size 0x10 lma 0x6c60
Start address 0x0000624a, load size 11376
Transfer rate: 19 KB/sec, 5688 bytes/write.
target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x0000058c msp: 0x2000d6a0
(gdb) c
Continuing.
^C
Program received signal SIGINT, Interrupt.
runtime.timerSleep (ticks=248) at /home/sago35/tinygo/tinygo/src/runtime/runtime_atsamd51.go:296
296             if timerWakeup.Get() != 0 {
```

Reply:
I have checked the following.

* samd51j19 : feather-m4
* samd51j20 : pyportal
* samd51p19 : wioterminal
* samd21g18 : xiao

I haven't been able to test the following because I don't have a board.

* samd21e18 : trinket-m0
* samd51g19 : itsybitsy-m4

Reply:
Okay, great! I just wanted to make sure it works for atsamd21, because of the different target.